nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -415,7 +416,24 @@ let
       echo "Image size: $image_size"
       truncate --size=$image_size "$out"
       faketime "2000-01-01 00:00:00" mkfs.vfat -i 12345678 -n EFIBOOT "$out"
-      mcopy -psvm -i "$out" ./EFI ./boot ::
+
+      # Force a fixed order in mcopy for better determinism, and avoid file globbing
+      for d in $(find EFI -type d | sort); do
```

Comment:
Maybe it's better to not repeat this and use `$(find EFI boot -type d | sort)` instead?

Reply:
I like your suggestion.